From fb519f0361565cad0d340095263a1e8866ec4f89 Mon Sep 17 00:00:00 2001
From: Alex Kiernan <alex.kiernan@gmail.com>
Date: Thu, 31 Oct 2019 11:30:00 +0000
Subject: [PATCH 1/3] tests: Handle EPIPE failures when head terminates

When using musl, it appears that the default is line buffered output, so
when `head -1` reads from a pipe we have to handle the source end of the
pipe getting EPIPE.

Signed-off-by: Alex Kiernan <alex.kiernan@gmail.com>
Upstream-Status: Accepted [v2019.6]
---
 tests/pull-test.sh            | 2 +-
 tests/test-pull-mirrorlist.sh | 4 ++--
 2 files changed, 3 insertions(+), 3 deletions(-)

diff --git a/tests/pull-test.sh b/tests/pull-test.sh
index 0a97a1195582..2cfd8e02f790 100644
--- a/tests/pull-test.sh
+++ b/tests/pull-test.sh
@@ -177,7 +177,7 @@ if ! skip_one_without_user_xattrs; then
     ${CMD_PREFIX} ostree --repo=cacherepo pull-local ostree-srv/gnomerepo main
     rev=$(ostree --repo=cacherepo rev-parse main)
     ${CMD_PREFIX} ostree --repo=cacherepo ls -R -C main > ls.txt
-    regfile_hash=$(grep -E -e '^-0' ls.txt | head -1 | awk '{ print $5 }')
+    regfile_hash=$((grep -E -e '^-0' ls.txt || true) | head -1 | awk '{ print $5 }')
     ${CMD_PREFIX} ostree --repo=repo remote add --set=gpg-verify=false corruptrepo $(cat httpd-address)/ostree/corruptrepo
     # Make this a loop so in the future we can add more object types like commit etc.
     for object in ${regfile_hash}.file; do
diff --git a/tests/test-pull-mirrorlist.sh b/tests/test-pull-mirrorlist.sh
index 4e44852d3971..85ff66e99f8e 100755
--- a/tests/test-pull-mirrorlist.sh
+++ b/tests/test-pull-mirrorlist.sh
@@ -47,12 +47,12 @@ setup_mirror content_mirror3
 
 # Let's delete a file from 1 so that it falls back on 2
 cd ${test_tmpdir}/content_mirror1/ostree/gnomerepo
-filez=$(find objects/ -name '*.filez' | head -n 1)
+filez=$((find objects/ -name '*.filez' || true) | head -n 1)
 rm ${filez}
 
 # Let's delete a file from 1 and 2 so that it falls back on 3
 cd ${test_tmpdir}/content_mirror1/ostree/gnomerepo
-filez=$(find objects/ -name '*.filez' | head -n 1)
+filez=$((find objects/ -name '*.filez' || true) | head -n 1)
 rm ${filez}
 cd ${test_tmpdir}/content_mirror2/ostree/gnomerepo
 rm ${filez}
-- 
2.17.1

